### ISSUE TITLE
TOFU dataset in pipeline

### ISSUE THREAD

**Issue by philswatton** [alan-turing-institute/arc-selective-forgetting#1]
- Add the TOFU dataset to the pipeline
- Implement the loading of the TOFU dataset in a format compatible with the [HuggingFace trainer](https://huggingface.co/docs/transformers/en/main_classes/trainer#transformers.Trainer.train_dataset) (e.g. [HF Datasets library dataset](https://huggingface.co/docs/datasets/v2.18.0/en/package_reference/main_classes#datasets.Dataset))
- Implement different levels of granularity for retain/forget sets (e.g. specific facts about authors vs authors)
- Implement fns for managing granularity in creating our experiments later on (i.e. we would like to be able to randomly select some data to drop at a specific level of granularity)

TOFU repo: <https://github.com/locuslab/tofu>

**Comment by philswatton** [comment]
For additional clarity, we want to have a single function to be loaded like:

```python
from arcsf import load_tofu 
tofu = load_tofu(kwargs)
```

or 
```python
import arcsf
tofu = arcsf.load_tofu(kwargs)
```

We want to be able to:

- Load the TOFU dataset with different forget sets conditional on arguments + random seed
- Load the TOFU dataset with a clear train/val/test split, with forget data being removed from the train set only
- Have an argument to define test sets of different granularity (forgetting facts vs authors)
- Have an argument to define/vary how *much* data of that given granularity is to be forgotten

Unit tests should be added to check the behaviour of these arguments

**Comment by J-Dymond** [comment]
I've made a start on this issue, and I've created a branch for it. 

So far I've created a function that is called like so:

```
from utils.data_utils import load_tofu

granularity = 'author_level' or 'fact_level_structured'

forget_set, retain_set = load_tofu(granularity = granularity, forgotten_author_fraction = ... , random_seed = ...)
```

Then it randomly chooses `forgotten_author_fraction` authors to remove data from, and depending on `granularity`, removes all questions to remove an author, or questions at the beginning of the QA list to remove biographical information regarding the author.
